spec/unit/csv_builder_spec.rb

```diff
@@ -233,6 +233,63 @@ def view_context
 
   end
 
+  context "build csv using specified encoding and encoding_options" do
+    let(:dummy_controller) {
+      class DummyController
+        def find_collection(*)
+          collection
+        end
+
+        def collection
+          Post
+        end
+
+        def apply_decorator(resource)
+          resource
+        end
+
+        def view_context
+        end
+      end
+      DummyController.new
+    }
+    let(:encoding) { Encoding::ASCII }
+    let(:opts) { {} }
+    let(:builder) do
+      ActiveAdmin::CSVBuilder.new(encoding: encoding, encoding_options: opts) do
+        column "おはようございます"
+        column "title"
+      end
+    end
+
+    context "Shift-JIS with options" do
+      let(:encoding) { Encoding::Shift_JIS }
+      let(:opts)     { {invalid: :replace, undef: :replace, replace: "?"} }
+
+      it "encodes the CSV" do
+        receiver = []
+        builder.build dummy_controller, receiver
+        line = receiver.last
+        expect(line.encoding).to eq(encoding)
+      end
+    end
+
+    context "ASCII with options" do
+      let(:encoding) { Encoding::ASCII }
+      let(:opts)     { {invalid: :replace, undef: :replace, replace: "__REPLACED__"} }
```

Comment:
Line is too long. [86/80]<br>Unnecessary spacing detected.<br>Space inside { missing.<br>Space inside } missing.